plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueToTrinoConverter.java

```diff
@@ -66,10 +66,17 @@ public static Database convertDatabase(com.amazonaws.services.glue.model.Databas
                 .build();
     }
 
-    public static Table convertTable(com.amazonaws.services.glue.model.Table glueTable, String dbName)
+    /**
+     * Convert the Glue table to the Hive representation, or empty if the table is missing necessary metadata.
+     * @param glueTable The source Glue table
+     * @param dbName The Glue database the table belongs to
+     * @return Empty if the table is missing required metadata, otherwise the Hive Table representation of the glueTable
+     */
+    public static Optional<Table> convertTable(com.amazonaws.services.glue.model.Table glueTable, String dbName)
     {
-        // TODO (https://github.com/trinodb/trino/issues/10902) glueTable.getStorageDescriptor() is an optional field in Glue
-        requireNonNull(glueTable.getStorageDescriptor(), () -> format("Table StorageDescriptor is null for table %s.%s (%s)", dbName, glueTable.getName(), glueTable));
+        if (glueTable.getStorageDescriptor() == null) {
+            return Optional.empty();
```

Comment:
Will this cause "table not found" to be returned when someone tries to access such a table?
This is confusing: user sees a table in SHOW TABLES, but then we tell them it doesn't exist.

Reply:
Yeah, it will. I have a test showing that here https://github.com/trinodb/trino/pull/10845/files#diff-07320a12f78f0c77af590dd6a46907f3c882bdef8e140b01a27ae8dd2b3bd718R155

I guess the other options would be to throw something like a `NotAHiveTableException` and handle that as a special case in the callers